From fa862fadc16580923e4a4075c0a7f15ae64061b9 Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Thu, 4 Feb 2021 10:24:57 +0200
Subject: [PATCH 57/78] s32: flexcan: Enable entire frame arbitration field
 comparison For Rx Mailboxes

If 'EACEN' bit is not set, extended ID frames will not be received.

Issue: ALB-6441
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
---
 drivers/net/can/flexcan.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/can/flexcan.c b/drivers/net/can/flexcan.c
index 38c6041d3b65..297f1794f8ff 100755
--- a/drivers/net/can/flexcan.c
+++ b/drivers/net/can/flexcan.c
@@ -6,7 +6,7 @@
 // Copyright (c) 2009 Sascha Hauer, Pengutronix
 // Copyright (c) 2010-2017 Pengutronix, Marc Kleine-Budde <kernel@pengutronix.de>
 // Copyright (c) 2014 David Jander, Protonic Holland
-// Copyright 2015,2018-2020 NXP
+// Copyright 2015,2018-2021 NXP
 //
 // Based on code originally by Andrey Volkov <avolkov@varma-el.com>
 
@@ -432,16 +432,16 @@ static const struct flexcan_devtype_data fsl_ls1021a_r2_devtype_data = {
 };
 
 static struct flexcan_devtype_data fsl_s32v234_devtype_data = {
-	.quirks = FLEXCAN_QUIRK_DISABLE_RXFG | FLEXCAN_QUIRK_DISABLE_MECR |
-		FLEXCAN_QUIRK_USE_OFF_TIMESTAMP |
+	.quirks = FLEXCAN_QUIRK_DISABLE_RXFG | FLEXCAN_QUIRK_ENABLE_EACEN_RRS |
+		FLEXCAN_QUIRK_DISABLE_MECR | FLEXCAN_QUIRK_USE_OFF_TIMESTAMP |
 		FLEXCAN_QUIRK_SETUP_STOP_MODE,
 	.n_irqs = ARRAY_SIZE(s32v234_flexcan_irqs),
 	.irqs = s32v234_flexcan_irqs,
 };
 
 static struct flexcan_devtype_data fsl_s32gen1_devtype_data = {
-	.quirks = FLEXCAN_QUIRK_DISABLE_RXFG | FLEXCAN_QUIRK_DISABLE_MECR |
-		FLEXCAN_QUIRK_USE_OFF_TIMESTAMP |
+	.quirks = FLEXCAN_QUIRK_DISABLE_RXFG | FLEXCAN_QUIRK_ENABLE_EACEN_RRS |
+		FLEXCAN_QUIRK_DISABLE_MECR | FLEXCAN_QUIRK_USE_OFF_TIMESTAMP |
 		FLEXCAN_QUIRK_SETUP_STOP_MODE,
 	.n_irqs = ARRAY_SIZE(s32gen1_flexcan_irqs),
 	.irqs = s32gen1_flexcan_irqs,
-- 
2.25.1

